### ISSUE TITLE
Reordering in Session Manager

### ISSUE THREAD

**Issue by igitur** [HeidiSQL/HeidiSQL#1935]
![image](https://github.com/HeidiSQL/HeidiSQL/assets/145854/5f155a60-cf71-4093-b610-9cab4d43166e)

**Is your feature request related to a problem? Please describe.**
I think the screenshot illustrates the issue. In the Session Manager, I want to reorder my items (either the folders, or the actual sessions), but it's proving very difficult or impossible. Sometimes, if I drag a folder, it ends up being a child folder of another folder. I struggle in particular to get the folder to the very first item. In the screenshot, I wanted to drag `0. Development` to be the very first item. 

At one stage my items were perfectly sorted as I wanted them. After an update, my sorting got messed up. Now I'm unable to get them sorted the way I wanted. 

My preference would actually be to use keyboard shortcuts to move them around, instead of drag 'n drop with the mouse, because it's proving just too difficult to hit the right spot.

But even with the drag 'n drop, the visual clues about where the item will end, isn't very clear. I'm not sure wither this is your control or a 3rd party control and what control you have over the control ( ;-) ), but I think the UX can be improved here.

**Comment by igitur** [comment]
Ok, I now understand the reasons for why it is as it is. The comment field is a very hacky workaround and will make reordering 20 sessions very tedious. Are you saying that this isn't possible with VirtualTree at all? If not, what about a custom, hidden sorting column that the user cann't see or edit, but which is populated with a sort order that is updated when a drag'n'drop occurs?

**Comment by ansgarbecker** [comment]
Of course that comment sorting is hacky. Just a thought. But think of the Windows explorer, or any other file manager. It does not have such a custom sorting through drag'n drop. Keep in mind that even after drag'n dropping, the HeidiSQL user may click on one of the column headers, which would break its manual order.

**Comment by igitur** [comment]
Yes, the combination of a treeview with listview-like column headers is a combination I haven't seen elsewhere, and creates its own nuances. I guess if I were to solve this, I would implement it like this:
- The treeview nodes have an intrinsic sort order that can be changed via drag 'n drop
- The column headers can sort the nodes. The column order can be: ascending, descending or unspecified and clicking on the column header cycles through these. This is similar to the column headers in the query results data grid.
- Only when all of the column headers are in the "unspecified" sort order does the intrinsic sort order of the treeview apply.

I think this could actually be solved with a hidden column quite easily, although that would also be considered a hacky solution.

**Comment by ansgarbecker** [comment]
Stumbling across this again, and I still think the "a" and "b" .. in the comment field does it quite well.
